Authentication_and_Authorization/hello-security/login-controller.php

```diff
@@ -6,19 +6,47 @@
 <body>
 <a href="index.php"> Take me home, country roads ... </a> <br>
 <?php
+
 include "db-connection.php";
+
+function getHashedPassword($conn, $email){
+
+    $sqlQueryForPassword = "SELECT password FROM person WHERE email = :email;";
+
+    $getHashedPassword = $conn->prepare($sqlQueryForPassword);
+    $getHashedPassword->bindParam(':email', $email);
+    $getHashedPassword->execute() or die ("Not valid credentials. There is no user with this email.");
+
+    $result = $getHashedPassword->fetch(PDO::FETCH_BOTH);
+    $hashedPassword = $result['password'];
+
+    return $hashedPassword;
+}
+
 $conn = openCon();
 
-$sql = "SELECT * from person where email = '" . htmlentities($_POST["email"]) . "' and password = '" . htmlentities($_POST["password"]) . "';";
+$email = htmlentities($_POST["email"]);
+$password = htmlentities($_POST["password"]);
+
+$hashedPassword = getHashedPassword($conn, $email);
+
+if(password_verify($password, $hashedPassword)){
+
+    $sql = "SELECT * from person WHERE email = :email;";
 
-$resultSet = $conn->query($sql) or die("Failed to query from DB!");
+    $resultSet = $conn->prepare($sql);
+    $resultSet->bindParam(':email', $email);
+    $resultSet->execute() or die("Failed to query from DB!");
+    $firstrow = $resultSet->fetch(PDO::FETCH_ASSOC) or die ("Not valid credentials.");
```

Comment:
Nice, this way you are not providing the user with information that the email is existing in the system.